x-pack/plugins/osquery/server/plugin.ts

```diff
@@ -21,6 +22,158 @@ import { initSavedObjects } from './saved_objects';
+
+const registerFeatures = (features: SetupPlugins['features']) => {
+  features.registerKibanaFeature({
+    id: 'osquery',
+    name: i18n.translate('xpack.features.osqueryFeatureName', {
+      defaultMessage: 'Osquery',
+    }),
+    order: 1300,
+    category: DEFAULT_APP_CATEGORIES.management,
+    app: ['osquery', 'kibana'],
+    catalogue: ['osquery'],
+    privileges: {
+      all: {
+        app: ['osquery', 'kibana'],
+        catalogue: ['osquery'],
+        savedObject: {
+          all: ['search', 'query'],
+          read: ['index-pattern'],
+        },
+        ui: ['show', 'save'],
+      },
+      read: {
+        app: ['osquery', 'kibana'],
+        catalogue: ['osquery'],
+        savedObject: {
+          all: [],
+          read: ['index-pattern', 'search', 'query'],
+        },
+        ui: ['show'],
+      },
+    },
+    subFeatures: [
+      {
+        name: i18n.translate('xpack.features.ossFeatures.discoverShortUrlSubFeatureName', {
+          defaultMessage: 'Live queries',
+        }),
+        privilegeGroups: [
+          {
+            groupType: 'mutually_exclusive',
+            privileges: [
+              {
+                id: 'live_queries_all',
+                includeIn: 'all',
+                name: 'All',
+                savedObject: {
+                  all: [],
+                  read: [],
+                },
+                ui: ['crudLiveQueries', 'readLiveQueries'],
+              },
+              {
+                id: 'live_queries_read',
+                includeIn: 'read',
+                name: 'Read',
+                savedObject: {
+                  all: [],
+                  read: [],
+                },
+                ui: ['readLiveQueries'],
+              },
+            ],
+          },
+          {
+            groupType: 'independent',
+            privileges: [
+              {
+                id: 'saved_queries_only',
+                name: i18n.translate(
+                  'xpack.features.ossFeatures.discoverCreateShortUrlPrivilegeName',
+                  {
+                    defaultMessage: 'Run Saved queries only',
+                  }
+                ),
+                includeIn: 'none',
+                savedObject: {
+                  all: [],
+                  read: [savedQuerySavedObjectType],
+                },
+                ui: ['savedQueriesOnly'],
+              },
+            ],
+          },
+        ],
+      },
+      {
+        name: i18n.translate('xpack.features.ossFeatures.discoverShortUrlSubFeatureName', {
+          defaultMessage: 'Saved queries',
+        }),
+        privilegeGroups: [
+          {
+            groupType: 'mutually_exclusive',
+            privileges: [
+              {
+                id: 'saved_queries_all',
+                includeIn: 'all',
+                name: 'All',
+                savedObject: {
+                  all: [savedQuerySavedObjectType],
+                  read: [savedQuerySavedObjectType],
```

Comment:
You don't need to grant both `all` and `read` SO privileges for a given type:
```suggestion
                  all: [savedQuerySavedObjectType],
                  read: [],
```